addons/sale_timesheet/models/project_update.py

```diff
@@ -4,59 +4,60 @@
 from odoo import api, models
 from odoo.tools import float_utils, format_amount, formatLang
 
+
 from collections import defaultdict
 
 class ProjectUpdate(models.Model):
     _inherit = 'project.update'
 
     @api.model
     def _get_template_values(self, project):
+        template_values = super(ProjectUpdate, self)._get_template_values(project)
+        services = self._get_services_values(project)
+        profitability = self._get_profitability_values(project)
+        show_sold = template_values['project'].allow_billable and 'data' in services and len(services['data']) > 0
+        show_profitability = profitability
         return {
             **super(ProjectUpdate, self)._get_template_values(project),
+            'show_sold': show_sold,
+            'show_profitability': show_profitability,
+            'show_activities': template_values['show_activities'] or show_sold or show_profitability,
             'services': self._get_services_values(project),
             'profitability': self._get_profitability_values(project),
         }
 
-    @api.model
-    def _get_project_sols(self, project):
-        return self.env['sale.order.line'].search([('order_id', '=', project.sale_order_id.id), ('is_service', '=', True)])
-
     @api.model
     def _get_services_values(self, project):
         if not project.allow_billable:
             return {}
         services = []
-        total_sold_per_uom = defaultdict(float)
-        total_effective_per_uom = defaultdict(float)
         total_sold, total_effective, total_remaining = 0, 0, 0
-        sols = self._get_project_sols(project)
+        sols = project._get_sale_order_lines()
+        name_by_sol = dict(sols.name_get())
         for sol in sols:
             #We only want to consider hours and days for this calculation
-            if sol.product_uom.category_id == self.env.company.timesheet_encode_uom_id.category_id:
+            if sol.product_uom.category_id == self.env.company.timesheet_encode_uom_id.category_id or sol.product_uom.name == "Units":
+                name = name_by_sol[sol.id] if len(sols.order_id) > 1 else sol.name
+                product_uom_qty = sol.product_uom._compute_quantity(sol.product_uom_qty, self.env.company.timesheet_encode_uom_id, raise_if_failure=False)
+                qty_delivered = sol.product_uom._compute_quantity(sol.qty_delivered, self.env.company.timesheet_encode_uom_id, raise_if_failure=False)
                 services.append({
-                    'name': sol.name,
-                    'uom': sol.product_uom.name,
-                    'sold_value': sol.product_uom_qty,
-                    'effective_value': sol.qty_delivered,
-                    'remaining_value': sol.product_uom_qty - sol.qty_delivered,
+                    'name': name,
```

Comment:
We can directly put the content of `name` variable in `name` key. 🙂 (not mandatory)
```suggestion
                product_uom_qty = sol.product_uom._compute_quantity(sol.product_uom_qty, self.env.company.timesheet_encode_uom_id, raise_if_failure=False)
                qty_delivered = sol.product_uom._compute_quantity(sol.qty_delivered, self.env.company.timesheet_encode_uom_id, raise_if_failure=False)
                services.append({
                    'name': name_by_sol[sol.id] if len(sols.order_id) > 1 else sol.name,
```